Cargo.toml

```diff
@@ -1,5 +1,46 @@
 [workspace]
 
 members = [
-   "rust/worker/"
+   "rust/blockstore",
+   "rust/cache",
+   "rust/config",
+   "rust/error",
+   "rust/storage",
+   "rust/types",
+   "rust/worker"
 ]
+
+[workspace.dependencies]
+serde = { version = "1.0.193", features = ["derive"] }
+serde_json = "1.0.108"
+arrow = "52.0.0"
+thiserror = "1.0.50"
+uuid = { version = "1.6.1", features = ["v4", "fast-rng", "macro-diagnostics"] }
+async-trait = "0.1.74"
+roaring = "0.10.3"
+futures = "0.3"
+parking_lot = "0.12.1"
+tracing = "0.1"
+tokio = { version = "1.0", features = ["macros", "rt-multi-thread"] }
+tokio-util = "0.7.10"
+tonic = "0.10"
+prost = "0.12"
+prost-types = "0.12"
+num_cpus = "1.16.0"
+flatbuffers = "24.3.25"
+
+chroma-blockstore = { path = "rust/blockstore" }
+chroma-error = { path = "rust/error" }
+chroma-config = { path = "rust/config" }
+chroma-storage = { path = "rust/storage" }
+chroma-cache = { path = "rust/cache" }
+chroma-types = { path = "rust/types" }
+worker = { path = "rust/worker" }
+
+# Dev dependencies
+"rand" = "0.8.5"
+rand_xorshift = "0.3.0"
+tempfile = "3.8.1"
+shuttle = "0.7.1"
+proptest = "1.4.0"
+proptest-state-machine = "0.1.0"
```

Comment:
this is an interesting pattern, I don't think I've seen it before for multi crate projects (not that I've seen a ton haha)
what's the advantage specifying versions here rather than inside each crate?

Reply:
Following this pattern makes it so that two crates can’t be dependent on different versions of the same module. Which could lead to strange behaviors